next.config.mjs

```diff
@@ -1,5 +1,32 @@
 // @ts-check
 
+/*
+ * Netlify UI does not support context-based environment variables.
+ * For example, we cannot configure XYZ=42 for preview releases (pull requests)
+ * and XYZ=4242 for production (main branch).
+ *
+ * As a workaround, we read Netlify’s CONTEXT env variable and then
+ * map PROD_XYZ / XYZ_PROD or PR_XYZ / XYZ_PR into XYZ.
+ */
+
+const suffixOrPrefix =
+  process.env.CONTEXT === "production"
+    ? "PROD"
+    : process.env.CONTEXT === "deploy-preview"
+    ? "PR"
+    : "";
+
+if (suffixOrPrefix) {
+  for (const key in process.env) {
+    if (key.startsWith(`${suffixOrPrefix}_`)) {
```

Comment:
Curious why do we need two ways of doing the same thing?
Can we just agree on prefix?
That would simplify the code, and make the naming more predicatble

Reply:
Yeah I agree with you that it’s best to support just one option. I’m still not sure which one will work best, so I’d like us to have some initial freedom. Netlify orders env vars alphabetically in their UI and I’m not sure whether prefix or suffix will be easier to use. My intuition says _suffix_, but [netlify-plugin-contextual-env](https://www.npmjs.com/package/netlify-plugin-contextual-env) uses prefix.

Once we’ve experimented and established the preferred, we can simplify this code a little.

Reply:
I'm good with suffix, let's find agreement on this.